### PR TITLE
Bump json version from 20080701 to 20170516

### PR DESCRIPTION
**Pull request by snleee**
1. upgrade the json version
2. fix the codes that are not backward compatible

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/utils/JSONUtil.java [hunk 1]
```diff
@@ -202,11 +201,6 @@ public Iterator keys() {
       return _inner.keySet().iterator();
     }
 
-    @Override
-    public Iterator sortedKeys() {
```
Comment (jenniferdai) [inline pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/utils/JSONUtil.java]: Do you know what this impacts/do query results change?

Comment (snleee) [reply pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/utils/JSONUtil.java]: This function never gets called. I will merge after testing.

Comment (Jackie-Jiang) [reply pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/utils/JSONUtil.java]: I think this whole class can be removed (along with PQLParserUtils)

File: pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/OfflineClusterIntegrationTest.java [hunk 2]
```diff
@@ -303,19 +304,19 @@ private void testNewAddedColumns() throws Exception {
     response = postQuery(pqlQuery);
     groupByResult =
         response.getJSONArray("aggregationResults").getJSONObject(0).getJSONArray("groupByResult").getJSONObject(0);
-    Assert.assertEquals(groupByResult.getInt("value"), 0);
+    Assert.assertEquals(groupByResult.getString("value"), "0.00000");
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/OfflineClusterIntegrationTest.java]: Let's use getDouble(), then convert to int or long

Comment (snleee) [reply pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/OfflineClusterIntegrationTest.java]: good catch. I was testing this and forgot to change.

**Pull request merged**